app/scripts/metamask-controller.test.js

```diff
@@ -748,29 +735,26 @@ describe('MetaMaskController', () => {
             allDetectedTokens: { '0x1': { [TEST_ADDRESS_2]: [{}] } },
           });
 
-        const startTime = Date.now();
         await metamaskController.createNewVaultAndRestore(
           'foobar1337',
           TEST_SEED,
         );
 
         // Expect first account to be selected
-        const identities = cloneDeep(metamaskController.getState().identities);
+        const accounts = cloneDeep(
+          metamaskController.accountsController.listAccounts(),
+        );
+
+        const selectedAccount =
+          metamaskController.accountsController.getSelectedAccount();
+
+        expect(selectedAccount.address).toBe(TEST_ADDRESS);
         expect(
-          identities[TEST_ADDRESS].lastSelected >= startTime &&
-            identities[TEST_ADDRESS].lastSelected <= Date.now(),
-        ).toStrictEqual(true);
-
-        // Expect first 2 accounts to be restored
-        delete identities[TEST_ADDRESS].lastSelected;
-        expect(identities).toStrictEqual({
-          [TEST_ADDRESS]: { address: TEST_ADDRESS, name: DEFAULT_LABEL },
-          [TEST_ADDRESS_2]: {
-            address: TEST_ADDRESS_2,
-            name: 'Account 2',
-            lastSelected: expect.any(Number),
-          },
-        });
+          accounts.find((account) => account.address === TEST_ADDRESS),
+        ).toBeDefined();
+        expect(
+          accounts.find((account) => account.address === TEST_ADDRESS_2),
+        ).toBeDefined();
```

Comment:
Based on the comment "Expect first account to be selected" and the original test, shouldn't we check `accounts[0]` and then `accounts[1]`.

Because using `.find` here could have a different order.

Reply:
Probably something like:
```ts
expect(selectedAccount.address).toBe(TEST_ADDRESS);
expect(accounts).toHaveLength(2);
expect(accounts[0].address).toBe(TEST_ADDRESS);
expect(accounts[0].metadata.name).toBe(DEFAULT_LABEL);
expect(accounts[1].address).toBe(TEST_ADDRESS_2);
expect(accounts[1].metadata.name).toBe('Account 2');
expect(accounts[1].metadata.lastSelected).toBeGreaterThan(accounts[0].metadata.lastSelected);
```